l3build-variables.lua

```diff
@@ -209,6 +209,24 @@ pvtext = pvtext or ".pvt"
 tlgext = tlgext or ".tlg"
 tpfext = tpfext or ".tpf"
 
+test_types = setmetatable(test_types or {}, { __index = {
+   log = {
+      test = lvtext,
+      generated = logext,
+      reference = tlgext,
+      expectation = lveext,
+      compare = compare_tlg,
+      rewrite = rewrite_log,
+   },
+   pdf = {
+      test = pvtext,
+      generated = pdfext,
+      reference = tpfext,
+      rewrite = rewrite_pdf,
+   },
+}})
+test_order = test_order or {"log", "pdf"}
+
```

Comment:
The default configuration. Maybe three interesting parts here: The split between `test_types` and `test_order`. Mostly it's done this way to ensure that behavior is deterministic in some edge cases, but it also allows to a switch to a auto loading schemes for additional formats later (To automatically fill test_types when types are used in test_order).

Names might need improvement.

Reply:
I'm not clear on what you are up to with metatables here :) Why not `test_types = test_types or  { }` then 
`test_types.log = test_types.log or { ... }` and so on?

Reply:
@josephwright No real reason, I'm just more used to the metatable approach. I changed it for consistency now.